audit-log-api/src/use-cases/GetLastEventUseCase.ts

```diff
@@ -1,20 +1,24 @@
 import type { AuditLogDynamoGateway, BichardAuditLogEvent, PromiseResult } from "shared"
 import { isError } from "shared"
 
-export default class GetLastEventUseCase {
+export default class GetLastFailedMessageEventUseCase {
   constructor(private readonly auditLogDynamoGateway: AuditLogDynamoGateway) {}
 
   async get(messageId: string): PromiseResult<BichardAuditLogEvent> {
-    const events = await this.auditLogDynamoGateway.fetchEvents(messageId)
+    const events = (await this.auditLogDynamoGateway.fetchEvents(messageId)) as BichardAuditLogEvent[]
 
     if (isError(events)) {
       return events
     }
 
-    if (events.length === 0) {
+    const failedEvents = events.filter(
+      (event) => event.category === "error" && event.s3Path && event.eventSourceQueueName
```

Comment:
Will the failed event _always_ have the keys `s3Path` and `eventSourceQueueName`?

Reply:
Yes, it should do. We won't be able to retry the failed event those keys!